enterprise/backend/test/metabase_enterprise/semantic_search/indexer_test.clj

```diff
@@ -184,18 +185,19 @@
             (semantic.indexer/on-indexing-idle indexing-state)
             (is (= 3000 @sleep-called))))))))
 
-(defn- open-loop-thread! [& loop-args]
+(defn- open-loop-thread! ^Closeable [& loop-args]
   (let [caught-ex (volatile! nil)]
     (semantic.tu/closeable
      {:caught-ex caught-ex
       :thread
       (doto (Thread.
              ^Runnable
-             (bound-fn []
-               (try
-                 (apply semantic.indexer/indexing-loop loop-args)
-                 (catch Throwable t
-                   (vreset! caught-ex t)))))
+             (identity
```

Comment:
super annoying, but `bound-fn` is one of the many macros that doesn't explicitely preserve the original form's meta (https://clojure.atlassian.net/browse/CLJ-865), so we need to wrap in identity to have a target we can attach the meta to

Reply:
switched to `bound-fn* (fn []` instead of `identity (bound-fn` as it's a bit less opaque